datatypes/src/primitives/time_interval.rs

```diff
@@ -167,7 +167,8 @@ impl TimeInterval {
     /// ```
     ///
     pub fn contains(&self, other: &Self) -> bool {
-        self.start <= other.start && self.end >= other.end
+        // TODO: should contains cover equals?
+        self == other || (self.start..self.end).contains(&other.start) && self.end >= other.end
```

Comment:
can you put parentheses around the seconds part so that it's clearer how this is evaluated?

Reply:
i can try later. I guess clippy will not be happy about that

Reply:
[0277db9](https://github.com/geo-engine/geoengine/pull/822/commits/0277db9c0d62b3af008c94b3d19382ff5eea9c8f)